docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,175 @@
+ - MklProxyNative
+   - Wrapper for Intel MKL. When Intel MKL is not present, this is not needed. However, the build is hardcoded to always compile this code.
+   - Small amount of work required to change the build process to exclude this as needed.
+ - SymSgdNative
+   - No managed fallback.
+   - Medium amount of work required to have a managed fallback. Only about 500 lines of code plus having to implement 4 vector operations.
+   - Small amount of work required port to other platforms. Only uses 4 Intel MKL methods which we can replace. I was only able to find 2 direct replacements on other architectures, but we could easily write our own for either those 2 or even all 4.
+   - This was created before we had hardware intrinsics so we used native code for performance as well as take advantage of IntelMKL.
+
+Of these 6, only LdaNative builds successfully for other architectures without changing anything.
+
+### 2.5 3rd Party Dependencies
+As mentioned above, there are several 3rd party packages that don't have support for non x86/x64 machines.
+ - LightGBM. LightGBM doesn't offer packages for non x86/x64. I was able to build the code for Arm64, but we would either have to build it ourselves, convince them to create more packages for us, or annotate that this doesn't work on non x86/64 machines.
+ - TensorFlow. The full version of TensorFlow only runs on x86/x64. There is a [lite](https://www.tensorflow.org/lite/guide/build_arm64) version that supports Arm64, and you can install it directly with python, but this isn't the full version so not all models will run. We would also have to verify if the C# library we use to interface with TensorFlow will work with the lite version.
+ - ONNX Runtime. ONNX Runtime doesn't have prebuilt packages for more than just x86/x64. It does support Arm, but we have to build it ourselves or get the ONNX Runtime team to package Arm assemblies. This is the same situation as with LightGBM.
+
+
+## 3 Possible Solutions
+We have several possible options we can use to resolve this:
+ - Keep ML.NET native components and rewrite them to avoid problematic dependencies completely.
+ - Keep ML.NET native components and ifdef/rewrite to avoid problematic dependencies on platforms/architectures only where they are not supported.
+ - Hybrid approach of replacement code and software fallback. This is the approach I recommend.
+
+None of these approaches resolve the 3rd party dependency issues. These solutions only deal with first party ML.NET code itself.
+
+I have lots more info about our dependency on x86/x64 in another document if required.
+
+### 3.1 Rewrite native components to work on other platforms
+This will still allow us to gain the benefits of the X86/x64 SIMD instructions and Intel MKL on architectures that support it but will also keep the benefits of native code in the other places. The downside is that we would have to build the native code for, potentially, a lot of different architectures.
```

Comment:
Or convince upstream to build it?

Reply:
Here I was meaning the native ML.NET components we wrote ourselves. No upstream in this case sadly.